index.js

```diff
@@ -176,6 +176,9 @@ function Fetch(url, opts) {
 
 			// prepare response
 			var body = res.pipe(new stream.PassThrough());
+
+			body.clientResponse = res;
```

Comment:
When decompression is requested, `body` is overwritten, so this `clientResponse` will get lost.

I was thinking of passing `req` as `clientRequest` through `response_options` into the `Response` constructor, and make `clientRequest` a property of the `Response` class. This way, not only would one be able to do:

```js
fetch('...').then(res => res.clientRequest.abort());
```

and cancel both the request and the response, which is especially helpful if chunked encoding is used.